app/models/manageiq/providers/vmware/cloud_manager.rb

```diff
@@ -0,0 +1,96 @@
+class ManageIQ::Providers::Vmware::CloudManager < ManageIQ::Providers::CloudManager
+  require_nested :RefreshParser
+  require_nested :RefreshWorker
+  require_nested :Refresher
+  require_nested :Vm
+
+  def self.ems_type
+    @ems_type ||= "vmware_vcd".freeze
+  end
+
+  def self.description
+    @description ||= "VMware vCloud Director".freeze
+  end
+
+  def self.default_blacklisted_event_names
+    []
+  end
+
+  def self.hostname_required?
+    true
+  end
+
+  def supports_port?
+    true
+  end
+
+  def supported_auth_types
+    %w(default)
+  end
+
+  def supports_authentication?(authtype)
+    supported_auth_types.include?(authtype.to_s)
+  end
+
+  #
+  # Connections
+  #
+
+  def self.raw_connect(server, port, username, password)
+    require 'fog/vcloud_director'
+
+    params = {
+      :vcloud_director_username      => username,
+      :vcloud_director_password      => password,
+      :vcloud_director_host          => server,
+      :vcloud_director_show_progress => false,
+      :port                          => port,
+      :connection_options            => {
+        :ssl_verify_peer => false # for development
+      }
+    }
+
+    Fog::Compute::VcloudDirector.new(params)
+  end
+
+  def connect(options = {})
+    raise "no credentials defined" if missing_credentials?(options[:auth_type])
+
+    server   = options[:ip] || address
+    port     = options[:port] || self.port
+    username = options[:user] || authentication_userid(options[:auth_type])
+    password = options[:pass] || authentication_password(options[:auth_type])
+
+    self.class.raw_connect(server, port, username, password)
+  end
+
+  def vcd_service
+    @vcd_service ||= connect
+  end
+
+  def translate_exception(err)
+    case err
+    when Fog::Compute::VcloudDirector::Unauthorized
+      MiqException::MiqHostError.new "Login failed due to a bad username or password."
```

Comment:
@bancek I think this would be more of a `MiqException::MiqInvalidCredentialsError` Exception